### PR TITLE
Use dict comprehension in plugins

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

This is PR mass-replace the construct `dict((k, v) for k, v in d if condition)` with dict comprehensions: `{k: v for k, v in d if condition}`

##### ISSUE TYPE

- Refactoring Pull Request

##### COMPONENT NAME

plugins/filter/hashids.py
plugins/filter/keep_keys.py
plugins/filter/remove_keys.py
plugins/filter/replace_keys.py
plugins/module_utils/csv.py
plugins/module_utils/mh/mixins/vars.py
plugins/module_utils/vardict.py
plugins/modules/apache2_mod_proxy.py
plugins/modules/gitlab_group.py
plugins/modules/keycloak_client.py
plugins/modules/keycloak_clientscope.py
plugins/modules/keycloak_identity_provider.py
plugins/modules/keycloak_user_federation.py
plugins/modules/linode.py
plugins/modules/lxd_container.py
plugins/modules/manageiq_provider.py
plugins/modules/one_service.py
plugins/modules/one_vm.py
plugins/modules/proxmox.py
plugins/modules/proxmox_disk.py
plugins/modules/proxmox_kvm.py
plugins/plugin_utils/unsafe.py
tests/sanity/extra/botmeta.py

### CODE REVIEW

File: plugins/modules/keycloak_clientscope.py [hunk 1]
```diff
@@ -428,7 +428,7 @@ def main():
         # Unfortunately, the ansible argument spec checker introduces variables with null values when
         # they are not specified
         if clientscope_param == 'protocol_mappers':
-            new_param_value = [dict((k, v) for k, v in x.items() if x[k] is not None) for x in new_param_value]
+            new_param_value = [{k: v for k, v in x.items() if x[k] is not None} for x in new_param_value]
```
Comment (russoz) [inline plugins/modules/keycloak_clientscope.py:431]:
```suggestion
            new_param_value = [{k: v for k, v in x.items() if v is not None} for x in new_param_value]
```

File: plugins/modules/keycloak_identity_provider.py [hunk 2]
```diff
@@ -534,7 +534,7 @@ def main():
     # special handling of mappers list to allow change detection
     if module.params.get('mappers') is not None:
         for change in module.params['mappers']:
-            change = dict((k, v) for k, v in change.items() if change[k] is not None)
+            change = {k: v for k, v in change.items() if change[k] is not None}
```
Comment (russoz) [inline plugins/modules/keycloak_identity_provider.py:537]:
```suggestion
            change = {k: v for k, v in change.items() if v is not None}
```

File: plugins/modules/keycloak_user_federation.py [hunk 3]
```diff
@@ -886,7 +886,7 @@ def main():
         new_param_value = module.params.get(param)
         old_value = before_comp[camel(param)] if camel(param) in before_comp else None
         if param == 'mappers':
-            new_param_value = [dict((k, v) for k, v in x.items() if x[k] is not None) for x in new_param_value]
+            new_param_value = [{k: v for k, v in x.items() if x[k] is not None} for x in new_param_value]
```
Comment (russoz) [inline plugins/modules/keycloak_user_federation.py:889]:
```suggestion
            new_param_value = [{k: v for k, v in x.items() if v is not None} for x in new_param_value]
```

File: plugins/modules/keycloak_user_federation.py [hunk 4]
```diff
@@ -895,7 +895,7 @@ def main():
         if module.params['provider_id'] in ['kerberos', 'sssd']:
             module.fail_json(msg='Cannot configure mappers for {type} provider.'.format(type=module.params['provider_id']))
         for change in module.params['mappers']:
-            change = dict((k, v) for k, v in change.items() if change[k] is not None)
+            change = {k: v for k, v in change.items() if change[k] is not None}
```
Comment (russoz) [inline plugins/modules/keycloak_user_federation.py:898]:
```suggestion
            change = {k: v for k, v in change.items() if v is not None}
```

File: plugins/modules/lxd_container.py [hunk 5]
```diff
@@ -748,19 +755,22 @@ def _apply_instance_configs(self):
...
             self.ignore_volatile_options = self.module.params.get('ignore_volatile_options')
 
             self.old_instance_json = self._get_instance_json()
-            self.old_sections = dict(
-                (section, content) if not isinstance(content, dict)
-                else (section, dict((k, v) for k, v in content.items()
-                                    if not (self.ignore_volatile_options and k.startswith('volatile.'))))
+            self.old_sections = {
+                section: adjust_content(content)
                 for section, content in (self.old_instance_json.get('metadata', None) or {}).items()
```
Comment (russoz) [inline plugins/modules/lxd_container.py:771]:
```suggestion
                for section, content in self.old_instance_json.get('metadata', {}).items()
```

**Comment by felixfontein** [comment]
Can you add a changelog fragment?

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks for improving the code!

**Pull request merged**